bin/cheritest/cheritest_fault.c

```diff
@@ -209,13 +209,16 @@ test_fault_cgetcause(const struct cheri_test *ctp __unused)
 void
 test_nofault_cfromptr(const struct cheri_test *ctp __unused)
 {
+	/* FIXME: uncomment once overloaded builtins have been merged */
+#ifdef NOTYET_THIS_NEEDS_OVERLOADED_BUILTINS_HAVE
 	char buf[256];
 	void * __capability cb; /* derived from here */
 	char * __capability cd; /* stored into here */
 
 	cb = cheri_ptr(buf, 256);
-	cd = __builtin_cheri_cap_from_pointer(cb, (void *)(uintptr_t)10);
```

Comment:
Currently dropping the `void *` gives:
```
warning: incompatible integer to pointer conversion passing '__uintcap_t __attribute__((cheri_no_provenance))' to parameter of type 'const void *' [-Wint-conversion]
```
So we can make this work with both by using a pragma to temporarily silence that warning and using the new style.

Reply:
Ah I assumed we were compiling with `-Werror=int-conversion`

Reply:
Warning or error, that's what the pragma would silence :)